src/AutoRest.CSharp.V3/Generation/Writers/CodeWriter.cs

```diff
@@ -392,14 +414,16 @@ public override string ToString()
                     .OrderByDescending(ns => ns.StartsWith("System"))
                     .ThenBy(ns=>ns)
                     .ToArray();
-
-            builder.AppendLine("// Copyright (c) Microsoft Corporation. All rights reserved.");
-            builder.AppendLine("// Licensed under the MIT License.");
-            builder.AppendLine();
-            builder.AppendLine("// <auto-generated/>");
-            builder.AppendLine();
-            builder.AppendLine("#nullable disable");
-            builder.AppendLine();
+            if (header)
```

Comment:
Helps testing.

Reply:
Right now, the header is written here because we assume ToString on a CodeWriter produces a file. How about we have something that represents a file that contains a CodeWriter? Then, we have this block be part of a file's starting text, and append the CodeWriter output to that. It is better encapsulation of concepts. Then, your tests don't need to call ToString differently.

Reply:
Yeah, I want to invert the logic a bit and have this handled in GeneratedCodeWorkspace as part of the formatting/cleanup.

Reply:
Sounds good. Can you create an issue so it doesn't get lost?

Reply:
https://github.com/Azure/autorest.csharp/issues/642